votes/forms.py

```diff
@@ -202,3 +226,125 @@ def save(self, request: HttpRequest, decision_id: int):
+        To delete an existing annotation, include the "delete" flag:
+        {
+            "person_id": 1234,
+            "delete": true
+        }
+        """,
+    )
+
+    def clean_annotations_json(self):
+        """
+        Validates the JSON data and converts it to a Pydantic model
+        """
+
+        json_data = self.cleaned_data.get("annotations_json")
+        if not json_data:
+            raise forms.ValidationError("This field is required.")
+
+        # Parse the JSON data
+        try:
+            annotations_data = json.loads(json_data)
+        except json.JSONDecodeError as e:
+            # Format the JSON error in a user-friendly way
+            line_col = (
+                f"line {e.lineno}, column {e.colno}"
+                if hasattr(e, "lineno")
+                else "unknown position"
+            )
+            error_msg = f"Invalid JSON format at {line_col}: {e.msg}"
+            raise forms.ValidationError(error_msg)
+
+        # Validate with Pydantic model
+        try:
+            vote_updates = VoteUpdates.model_validate(annotations_data)
+        except ValidationError as e:
+            # Format validation errors in a user-friendly way
+            error_messages = []
+            for error in e.errors():
+                location = ".".join(str(loc) for loc in error["loc"])
+                message = error["msg"]
+                error_messages.append(f"Error at {location}: {message}")
+
+            formatted_error = (
+                "Please fix the following validation errors:\n"
+                + "\n".join(error_messages)
+            )
+            raise forms.ValidationError(formatted_error)
+
+        # Return the parsed and validated data so it's available in save
+        return vote_updates
+
+    def save(self, request: HttpRequest, decision_id: int):
+        """
+        Process the validated data and create/update/delete annotations
+        This method assumes all validation has already happened in clean method
+        """
+
+        # Get the validated vote updates from cleaned_data
+        vote_updates = self.cleaned_data["annotations_json"]
+
+        # Process each annotation update
+        created_count = 0
+        updated_count = 0
+        deleted_count = 0
+
+        for update in vote_updates.root:
+            # Check if this annotation should be deleted
+            if update.delete:
+                try:
+                    annotation = VoteAnnotation.objects.get(
+                        division_id=decision_id, person_id=update.person_id
+                    )
+                    annotation.delete()
+                    deleted_count += 1
+                except VoteAnnotation.DoesNotExist:
+                    # If it doesn't exist, nothing to delete
+                    pass
+            else:
+                # Try to find existing annotation for this person and division
+                annotation, created = VoteAnnotation.objects.update_or_create(
```

Comment:
if you have an incorrect person id then this creates an annotation with a blank person id that is then, surprisingly, not accessible in the admin and there's no warning anywhere.